CodingConventions.md

```diff
@@ -0,0 +1,307 @@
+
+For example, [`VoteAggregator`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L29) is
+a data flow vertex within a consensus nodes, whose job it is to collect incoming votes from other consensus nodes.
+Hence, the corresponding function [`AddVote(vote *model.Vote)`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L37)
+treats all inputs as untrusted. Furthermore, `VoteAggregator` consumes auxiliary inputs
+([`AddBlock(block *model.Proposal)`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L45),
+[`InvalidBlock(block *model.Proposal)`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L51),
+[`PruneUpToView(view uint64)`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L57))
+from the node's main consensus logic, which trusted.
+
+
+### Byzantine Inputs result in benign sentinel errors
+
+Any input that is coming from another node is untrusted and the vertex processing it should gracefully handle any arbitrarily malicious input.
+Under no circumstances should a byzantine input result in the vertex's internal state being corrupted.
+Per convention, failure case due to a byzantine inputs are represented by specifically-typed
+[sentinel errors](https://pkg.go.dev/errors#New) (for further details, see error handling section).
+
+Byzantine inputs are one particular case, where a function returns a 'benign error'. The critical property for an error
+to be benign is that the component returning it is still fully functional, despite encountering the error condition. All
+benign errors should be handled within the vertex.
+
+## Error handling
+
+This is a highly-compressed summary
+of [Error Handling in Flow (Notion)](https://www.notion.so/dapperlabs/Error-Handling-in-Flow-a7692d60681c4063b48851c0dbc6ed10)
+
+A blockchain is a security-first system. Therefore, encountering a problem and continuing on a "best effort" basis is
+*not* an option. There is a clear definition of the happy path (i.e. the block must be in storage). Any deviation of this
+happy path is either
+
+1. a protocol violation (from a different node), which itself is clearly defined
+2. is an uncovered edge case (which could be exploited in the worst case to compromise the system)
+3. a corrupted internal state of the node
+
+### Best Practice Guidelines
+
+1. **Errors are part of your API:** If there is an error return, there is a documentation what the error means. We conceptually differentiate between the following two classes of errors:
```

Comment:
suggest: "If there is an error returned the meaning of the error is clearly documented in the code"